drivers/mspi/mspi_nrfe.c

```diff
@@ -473,8 +473,12 @@ static int xfer_packet(struct mspi_xfer_packet *packet, uint32_t timeout)
 
 	/* Wait for the transfer to complete and receive data. */
 	if ((packet->dir == MSPI_RX) && (ipc_receive_buffer != NULL) && (ipc_received > 0)) {
-		memcpy((void *)packet->data_buf, (void *)ipc_receive_buffer, ipc_received);
-		packet->num_bytes = ipc_received;
+		/*
+		 * Writing to packet->num_bytes causes memory access errors, so for now
+		 * number of received data is stored in the first byte in data buffer.
+		 */
+		packet->data_buf[0] = ipc_received;
+		memcpy((void *)(packet->data_buf + 1), (void *)ipc_receive_buffer, ipc_received);
```

Comment:
```suggestion
		if(packet->num_bytes != ipc_received)
		{
			ipc_receive_buffer = NULL;
			ipc_received = 0;
			return -EIO;
		}
		memcpy((void *)(packet->data_buf + 1), (void *)ipc_receive_buffer, ipc_received);
```

Reply:
Or better, 
```suggestion
		if(packet->num_bytes != ipc_received)
		{
			rc = -EIO;
		}
		else
		{
			memcpy((void *)(packet->data_buf + 1), (void *)ipc_receive_buffer, ipc_received);
		}
```
Then cleanup will be done for both cases.